src/components/input-switch/array-field.tsx

```diff
@@ -1,108 +1,326 @@
+import '@isrd-isi-edu/chaise/src/assets/scss/_array-field.scss';
+
 // components
-import ClearInputBtn from '@isrd-isi-edu/chaise/src/components/clear-input-btn';
-import InputField, { InputFieldProps } from '@isrd-isi-edu/chaise/src/components/input-switch/input-field';
-//hooks
-import { useRef } from 'react';
+import { InputFieldProps } from '@isrd-isi-edu/chaise/src/components/input-switch/input-field';
+
 // utils
-import { getSimpleColumnType } from '@isrd-isi-edu/chaise/src/utils/input-utils';
+import InputSwitch from '@isrd-isi-edu/chaise/src/components/input-switch/input-switch';
 import { dataFormats } from '@isrd-isi-edu/chaise/src/utils/constants';
-import { arrayFieldPlaceholder } from '@isrd-isi-edu/chaise/src/utils/input-utils';
-import { windowRef } from '@isrd-isi-edu/chaise/src/utils/window-ref';
-import { hasVerticalScrollbar } from '@isrd-isi-edu/chaise/src/utils/input-utils';
+import { formatDatetime, formatFloat, formatInt, getInputType } from '@isrd-isi-edu/chaise/src/utils/input-utils';
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  DragDropContext, Draggable, DraggableProvided, Droppable, DroppableProps, DroppableProvided, DropResult
+} from 'react-beautiful-dnd';
+import { EventType, useFormContext, useFormState, useWatch } from 'react-hook-form';
+
+// since we're using strict mode, react-beautiful-dnd misbehaves due to multiple renders caused by strict mode
+// this is to guard against it
+const StrictModeDroppable = React.memo(function StrictModeDroppable({ children, ...props }: DroppableProps) {
```

Comment:
Move this outside of this file. It should go to its own file under `src/components/chaise-droppable.tsx` and let's call the component `ChaiseDroppable`.